airbyte-integrations/connectors/source-intercom/source_intercom/components.py

```diff
@@ -0,0 +1,341 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import re
+from dataclasses import InitVar, dataclass, field
+from typing import Any, Iterable, List, Mapping, Optional, Union
+
+import dpath.util
+import requests
+from airbyte_cdk.models import AirbyteMessage, SyncMode, Type
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from airbyte_cdk.sources.declarative.requesters.error_handlers.backoff_strategies.header_helper import get_numeric_value_from_header
+from airbyte_cdk.sources.declarative.requesters.error_handlers.backoff_strategy import BackoffStrategy
+from airbyte_cdk.sources.declarative.requesters.request_option import RequestOptionType
+from airbyte_cdk.sources.declarative.stream_slicers.stream_slicer import StreamSlicer
+from airbyte_cdk.sources.declarative.stream_slicers.substream_slicer import ParentStreamConfig
+from airbyte_cdk.sources.declarative.types import Config, Record, StreamSlice, StreamState
+from airbyte_cdk.sources.streams.core import Stream
+from dataclasses_jsonschema import JsonSchemaMixin
+
+
+@dataclass
+class IncrementalSingleSlice(StreamSlicer, JsonSchemaMixin):
```

Comment:
can you create issues for custom components as described [here](https://docs.google.com/document/d/1LB4gBk0slHj3zNmq1Qtj1U7VkKyTN9eFsVldgQ8T4HQ/edit?usp=drivesdk)

Reply:
On my way!